Add `log-level` mount option support for glusterfs plugin

humblec opened the pull request:
The glusterfs plugin/driver set log-level to ERROR by default while
mounting glusterfs shares. However at times, its required to supply other
log-level options like INFO, TRACE..etc. This patch enables the support
to provide other log-level values from storageclass.

Additional Ref#
https://docs.gluster.org/en/v3/Administrator%20Guide/Setting%20Up%20Clients/

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
4. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note

```

humblec commented:
/release-note-none

humblec commented:
/assign @rootfs @jingxu97

humblec commented:
/assign @wongma7

humblec commented on pkg/volume/glusterfs/glusterfs.go:
```diff
@@ -323,7 +330,10 @@ func (b *glusterfsMounter) setUpAtInternal(dir string) error {
 
 	// Use derived/provided log file in gluster fuse mount
 	options = append(options, "log-file="+log)
```
@linyouchong https://github.com/kubernetes/kubernetes/pull/68435 should address it.

humblec replied:
@linyouchong did that answer ? or the query was different ? :)

linyouchong replied:
@humblec Yes, I have no more question:)
I didn't saw #68435.

warmchang replied:
> @linyouchong #68435 should address it.

lgtm

humblec commented:
@jingxu97 Can you please review this ?

humblec commented:
/assign @gnufied

humblec commented:
/assign @jsafrane

jsafrane commented:
/lgtm

warmchang commented:
@humblec , Looks like needs to add the release note or tag release-note-none.

humblec commented:
/release-note-none

nikhita commented:
/kind feature

humblec committed:
Add `log-level` mount option support for glusterfs plugin

The glusterfs plugin/driver set log-level to ERROR by default while
mounting glusterfs shares. However at times, its required to supply other
log-level options like INFO, TRACE..etc. This patch enables the support
to provide other log-level values from storageclass.

Additional Ref#
    https://docs.gluster.org/en/v3/Administrator%20Guide/Setting%20Up%20Clients/

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec commented:
/retest

nikhita commented:
/retest

nikhita commented:
/retest

On Wed, Oct 3, 2018 at 7:44 PM k8s-ci-robot <notifications@github.com>
wrote:

> @humblec <https://github.com/humblec>: The following test *failed*, say
> /retest to rerun them all:
> Test name Commit Details Rerun command
> pull-kubernetes-integration fdcd58a
> <https://github.com/kubernetes/kubernetes/commit/fdcd58a794d926495eade54c5221609af50ea51b>
> link
> <https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/68440/pull-kubernetes-integration/29677/> /test
> pull-kubernetes-integration
>
> Full PR test history <https://k8s-gubernator.appspot.com/pr/68440>. Your
> PR dashboard <https://k8s-gubernator.appspot.com/pr/humblec>. Please help
> us cut down on flakes by linking to
> <https://git.k8s.io/community/contributors/devel/flaky-tests.md#filing-issues-for-flaky-tests>
> an open issue
> <https://github.com/kubernetes/kubernetes/issues?q=is:issue+is:open> when
> you hit one in your PR.
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/68440#issuecomment-426652292>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/APXA0Ok2lkYs3jt9aLZKpNa5yNY-3pHpks5uhMZSgaJpZM4WgK2K>
> .
>

humblec commented:
@jsafrane Could you please bring back lgtm here ?

jsafrane commented:
/lgtm
